src/commands/team.rs

```diff
@@ -1,163 +1,690 @@
+                Ok(seasons) => {
+                    self.current_season = Some(seasons.data[0].id);
+                    Some(seasons.data)
+                },
+                Err(err) => {
+                    return CreateInteractionResponseMessage::new()
+                        .add_embed(
+                            CreateEmbed::new().title("Failed to get season information about team from RobotEvents.")
+                            .description(format!("```rs\n{err:?}```")),
+                        );
+                }
+            };
+
+            // This check is technically redundant, since we just set `self.current_page` to Some(T), but
+            // it keeps us from have to unnecessarily clone using other methods, so... ¯\_(ツ)_/¯
+            CreateInteractionResponseMessage::new()
+                .embed(self.embed(self.current_page, robotevents, vrc_data_analysis, skills_cache).await)
+                .components(self.components(self.current_page, self.current_season.unwrap()))
+        } else {
+            CreateInteractionResponseMessage::new().content("Failed to find a RobotEvents team with this number.")
+        }
+    }
+
+    /// Returns the RobotEvents team data associated with this instance of [`Self`].
+    /// 
+    /// If `self.team` happens to be `None`, this function will attempt to fetch the required information
+    /// from the RobotEvents API, but otherwise return the cached result.
+    pub async fn find_robotevents_team(
+        &mut self,
+        robotevents: &RobotEvents,
+    ) -> Result<Team, TeamCommandRequestError> {
+        // Ensure that a team number has been provided by a user.
+        let team_number = if let Some(team_number) = &self.team_number {
+            team_number
+        } else {
+            // This can only be `None` if the command has not been responded to at all, or the bot somehow
+            // updates an invalid response which shouldn't realistically happen.
+            return Err(TeamCommandRequestError);
+        };
+
+        // If we've already fetched the team data once (such as in the case of page changes with message edits),
+        // use the data already stored in [`Self`].
+        if let Some(team) = self.team.clone() {
+            return Ok(team);
+        }
+
+        // Fetch team using RobotEvents HTTP client
+        let mut query = TeamsQuery::new().number(team_number.to_string());
+        if let Some(program_id_filter) = self.program_id_filter {
+            query = query.program(program_id_filter);
+        }
+
+        if let Ok(teams) = robotevents.teams(query).await {
+            if let Some(team) = teams.data.iter().next() {
+                self.team = Some(team.clone()); // Cache value for later use. 
+                return Ok(team.clone());
             }
+        }
+
+        Err(TeamCommandRequestError)
+    }
+    
+    /// Responds to an interaction with the `/team` command's message components (e.g. select menus).
```

Comment:
This comment does not at all describe what this function actually does. Again, your comment in the Discord is much more helpful.